### ISSUE TITLE
Update an element when another one is updated (DOC)

### ISSUE THREAD

**Issue by pmithrandir** [FriendsOfSymfony/FOSElasticaBundle#824]
Hello,

I think it would be interesting to provide a documentation explaining how to customize the update / insert / delete event listener between doctrine and ES.

It is said that it's not a good idea for performance to let the system in automatic mode, but there is not so much explanation on how to do it better.

Also, a common case could be to trigger the update on a foreign key when changing / adding an element.

Example : 
I create a comment that refer to a post.
A short description on how to trigger the update on the post from the comment would be interesting.

On a global point of view, More context should be given when documenting something. 
This page : https://github.com/FriendsOfSymfony/FOSElasticaBundle/blob/master/Resources/doc/cookbook/custom-properties.md
Might be an answer to my question, but without knowing what is what, it's really hard to understand what you try to achieve in the code. A functional description of the goal + a quick overview of the object / mapping involved could really help to understand what you are doing.

**Comment by merk** [comment]
There is some work started for cascading the update scheduling when a foreign key is modified however it turns out it isnt very easy to achieve in a generic way.

https://github.com/FriendsOfSymfony/FOSElasticaBundle/pull/681 started with this work, however only going based off if a collection changes, and in this case it doesnt work because the owner of a collection is marked dirty when the collection changes.

What needs to happen is we need to introduce a configuration for each type that tells us how to discover if a relationship has changed and I havent been able to think of a good way of doing this until we move to annotation/yml based mapping like the ORM.

As for the documentation visibility: yes, its a problem. However, I'm not the best at writing it and would appreciate assistance from anyone willing to spend some time to tidy it up, make sure it makes sense and in general improve the quality of it.

**Comment by pmithrandir** [comment]
Hello,

After a break on my project, I'm back on the same issue. 
I have 2 entities, one indexed in ES, the other one not.
When I update the second, I expect to update the first index to change the number of sub element.

Exemple : 
I index all my post, and all of them might have comment.

I want to be able to know and to order my post by number of comment.
I need either to store all my comment as nested element and to find a way to sort by their number(not sure if possible) or to store only the number, but to update it.

Any way to force update on an element in the index maybe ? I can definitly call a function to do that when saving a comment.

Pierre

**Comment by merk** [comment]
You can schedule an object for reindexing by calling https://github.com/FriendsOfSymfony/FOSElasticaBundle/blob/master/Persister/ObjectPersisterInterface.php#L28 on the appropriate persister service. Its id will be fos_elastica.object_persister.<index>.<type>

**Comment by pmithrandir** [comment]
Hello,

Persister service ? I don't think I heard about this one already.
Do you have any example of code ?
I know how to get the ES index object, but outside of that, I don't think anything else was present in documentation.

Pierre

**Comment by pmithrandir** [comment]
My apologies, I realized what you were trying to say.

I created the following command in my Post Controller.
```
    private function updateES ($post){
    
        $persister = $this->get('fos_elastica.object_persister.jaiuneidee.post');
        $persister->insertOne($post);
    }
```
And then, it was only a matter of calling this method after adding / removing a comment.

Attention, this command will use the current $post object.
So, if you have something like 
```
$this->updateES ($comment->getPost());
```
Be careful of 2 thing : 
 - you must flush your comment before.
 - you must add your comment to your comments variable inside your Post object. 

Otherwise, you will analyze the situation without the last comment, which is not what you want here.

Thank you, it work like a charm now.
Pierre
